Dependencies are not exposed to maven central causing build failure

Hello there,

At SW360, we are using `org.spdx:tools-java:2.0.0` from maven central. However, I cannot see any dependency defined in project's `pom.xml` on central: https://central.sonatype.com/artifact/org.spdx/tools-java/2.0.0/dependencies

This causes our test cases to fail due to missing classes of various stores. Can you please help us with it?

```
java.lang.NoClassDefFoundError: org/spdx/spdxRdfStore/RdfStore
	at org.spdx.tools.SpdxToolsHelper.fileTypeToStore(SpdxToolsHelper.java:100)
	at org.spdx.tools.SpdxToolsHelper.deserializeDocumentCompatV2(SpdxToolsHelper.java:211)
	at org.eclipse.sw360.spdx.SpdxBOMImporter.openAsSpdx(SpdxBOMImporter.java:134)
```

If I add `org.spdx:spdx-rdf-store:2.0.0` as dependency of my `pom.xml`, Java start complaining about `java.lang.NoClassDefFoundError: org/spdx/v3jsonldstore/JsonLDStore`

Link to PR: https://github.com/eclipse-sw360/sw360/pull/3095

Failing test cases: https://github.com/eclipse-sw360/sw360/actions/runs/14570385211/job/40866454627?pr=3095#step:12:912

Hey @goneall , I had to add the `spdx-java-library` as a dependency to my `pom.xml` as `tools-java` only reports junit as its dependency and I start getting symbol missing errors:
```
[ERROR] /home/z003u7px/development/sw360/backend/common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java:[73,21] package org.spdx.core does not exist
[ERROR] /home/z003u7px/development/sw360/backend/common/src/main/java/org/eclipse/sw360/spdx/SpdxBOMImporter.java:[36,21] package org.spdx.core does not exist
[ERROR] /home/z003u7px/development/sw360/backend/common/src/main/java/org/eclipse/sw360/spdx/SpdxBOMImporter.java:[37,21] package org.spdx.core does not exist
[ERROR] /home/z003u7px/development/sw360/backend/common/src/main/java/org/eclipse/sw360/spdx/SpdxBOMImporter.java:[38,46] package org.spdx.library.model.v2.enumerations does not exist
```

I am attaching the output of `$ mvn clean package -Dbase.deploy.dir=. -DskipTests=true -X -U` as requested for you to look at: [logs.txt](https://github.com/user-attachments/files/19866146/logs.txt)

Even `mvn dependency:tree` says `org.spdx:tools-java:jar:2.0.0:compile` has no transitive dependency.
```
$ mvn dependency:tree
Picked up _JAVA_OPTIONS: -Djdk.net.URLClassPath.disableClassPathURLCheck=true -Djdk.util.zip.disableZip64ExtraFieldValidation=true
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.eclipse.sw360:backend-common >------------------
[INFO] Building backend-common 19.2.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ backend-common ---
[INFO] org.eclipse.sw360:backend-common:jar:19.2.0
[INFO] +- org.eclipse.sw360:backend-service-core:jar:19.2.0:compile
[INFO] +- org.spdx:tools-java:jar:2.0.0:compile
[INFO] +- org.cyclonedx:cyclonedx-core-java:jar:10.2.1:compile
[INFO] |  +- commons-codec:commons-codec:jar:1.18.0:compile
[INFO] |  +- commons-io:commons-io:jar:2.18.0:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.17.0:compile
[INFO] |  +- org.apache.commons:commons-collections4:jar:4.4:compile
[INFO] |  +- com.github.package-url:packageurl-java:jar:1.5.0:compile
[INFO] |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-xml:jar:2.18.3:compile
[INFO] |  |  +- org.codehaus.woodstox:stax2-api:jar:4.2.2:compile
[INFO] |  |  \- com.fasterxml.woodstox:woodstox-core:jar:7.0.0:compile
[INFO] |  \- com.networknt:json-schema-validator:jar:1.5.6:compile
[INFO] |     +- com.ethlo.time:itu:jar:1.10.3:compile
[INFO] |     \- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.18.1:compile
[INFO] |        \- org.yaml:snakeyaml:jar:2.3:compile
..
Chopped for brevity.
..
[INFO] +- org.eclipse.sw360:datahandler:test-jar:tests:19.2.0:test
[INFO] +- org.eclipse.sw360:build-configuration:jar:19.2.0:compile
[INFO] +- org.eclipse.sw360:build-configuration:test-jar:tests:19.2.0:test
[INFO] \- jakarta.mail:jakarta.mail-api:jar:2.0.1:compile
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.579 s
[INFO] Finished at: 2025-04-23T16:47:08+05:30
[INFO] ------------------------------------------------------------------------
```

--- a/pom.xml
+++ b/pom.xml
@@ -285,7 +285,8 @@
 		    <plugin>
 			    <groupId>org.apache.maven.plugins</groupId>
 			    <artifactId>maven-shade-plugin</artifactId>
-				<version>3.4.1</version>
+				<version>3.2.4</version>
+				<!-- DO NOT USE LATER VERSIONS OF THE SHADE PLUGIN - SEE ISSUE 201 FOR CONTEXT -->
 			    <configuration>
 			        <shadedArtifactAttached>true</shadedArtifactAttached>
 			        <shadedClassifierName>jar-with-dependencies</shadedClassifierName>

